compiler/damlc/lib/DA/Cli/Damlc/Packaging.hs

```diff
@@ -110,23 +96,16 @@ createProjectPackageDb projectRoot (disableScenarioService -> opts) (PackageSdkV
       let dependenciesInPkgDbIds =
               Set.fromList $ map LF.dalfPackageId $ MS.elems dependenciesInPkgDb
 
-      -- Now handle data-dependencies.
-      darsFromDataDependencies <- getDarsFromDataDependencies dependenciesInPkgDbIds dataDeps
-      let dalfsFromDataDependencies = concatMap dalfs darsFromDataDependencies
-
-      -- All transitive packages from DARs specified in  `dependencies`.
-      -- This is only used for unit-id collision checks
-      -- and dependencies on newer LF versions.
-      darsFromDependencies <- getDarsFromDependencies dependenciesInPkgDbIds depsExtracted
-      let dalfsFromDependencies = concatMap dalfs darsFromDependencies
+      -- This is only used for unit-id collision checks and dependencies on newer LF versions.
+      dalfsFromDependencies <- queryDalfsFromDependencies depsDir dependenciesInPkgDbIds
+      dalfsFromDataDependencies <- queryDalfsFromDataDependencies depsDir dependenciesInPkgDbIds
+      mainDalfs <- queryMainDalfs depsDir dependenciesInPkgDbIds
 
       let dependencyInfo = DependencyInfo
               { dependenciesInPkgDb
               , dalfsFromDependencies
               , dalfsFromDataDependencies
-              , mainUnitIds =
-                    map (decodedUnitId . mainDalf)
-                        (darsFromDataDependencies ++ darsFromDependencies)
+              , mainUnitIds = map decodedUnitId mainDalfs
```

Comment:
Let’s leave it from this PR but the checks below for inconsistent versions, incompatible lf versions & unit id conflicts look like we can already do them when setting up `.daml/dependencies` instead of only failing here.

Reply:
it's a bit tricky because you need to know the unit id's as registered in the package database for these checks.

Reply:
What’s stopping us from figuring out the unit ids during the install phase?

Reply:
you can do that. I like the idea of the dependencies db to be completely independent of the tricks we need to apply to the package database.